tests/operators/test_agnostic_boolean_check.py

```diff
@@ -53,194 +53,114 @@ def drop_table_snowflake(
-                    conn_id="snowflake_conn",
-                    table_name=self.snowflake_table,
-                    schema=os.getenv("SNOWFLAKE_SCHEMA"),
-                    database=os.getenv("SNOWFLAKE_DATABASE"),
-                    warehouse=os.getenv("SNOWFLAKE_WAREHOUSE"),
-                ),
-                checks=[
-                    Check("test_1", " rooms > 7"),
-                    Check("test_2", " beds >= 3"),
-                ],
-                max_rows_returned=10,
-            )
-            a.execute({"run_id": "foo"})
-            assert False
-        except ValueError:
-            assert True
-
-    def test_happyflow_bigquery_success(self):
-        try:
-            a = boolean_check(
-                table=Table(
-                    self.table,
-                    conn_id="bigquery",
-                    schema=SCHEMA,
-                ),
-                checks=[Check("test_1", "rooms > 3")],
-                max_rows_returned=10,
-            )
-            a.execute({"run_id": "foo"})
-            assert True
-        except ValueError:
-            assert False
-
-    def test_happyflow_sqlite_success(self):
-        try:
-            a = boolean_check(
-                table=self.table_sqlite,
-                checks=[Check("test_1", "rooms > 3")],
-                max_rows_returned=10,
-            )
-            a.execute({"run_id": "foo"})
-            assert True
-        except ValueError:
-            assert False
-
-    def test_happyflow_bigquery_fail(self):
-        try:
-            a = boolean_check(
-                table=Table(
-                    self.table,
-                    conn_id="bigquery",
-                    schema=SCHEMA,
-                ),
+    run_dag(sample_dag)
+
+
+@pytest.mark.parametrize("table", SUPPORTED_DATABASES, indirect=True)
+def test_happyflow_fail(sample_dag, table):
+    @aql.transform
+    def get_table(input_table: Table):
+        return "SELECT * FROM {{input_table}}"
+
+    with pytest.raises(BackfillUnfinished):
+        with sample_dag:
+            temp_table = get_table(table)
+            aql.boolean_check(
+                table=temp_table,
                 checks=[
                     Check("test_1", "rooms > 7"),
                     Check("test_2", "beds >= 3"),
                 ],
                 max_rows_returned=10,
             )
-            a.execute({"run_id": "foo"})
-            assert False
-        except ValueError:
-            assert True
+        run_dag(sample_dag)
+
+
+@pytest.mark.parametrize(
+    "table",
+    [
+        "postgres",
+        pytest.param(
+            "bigquery",
+            marks=pytest.mark.xfail(
+                reason="bigquery don't expect table name before cols."
+            ),
+        ),
+        pytest.param(
+            "snowflake",
+            marks=pytest.mark.xfail(
+                reason="Binding data in type (table) is not supported."
+            ),
+        ),
```

Comment:
Can you please explain further why this test is expected to fail in snowflake?

Reply:
I think it's a bug, I"ll explore it as part of a new ticket.

Reply:
@utkarsharma2 could you create and link the two tickets - one for `BigQuery` and one for `snowflake`, before merging this PR?

Reply:
Created it - https://github.com/astro-projects/astro/issues/219